multinode-demo/start_nodes.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Format for the nodes file
+# node1 public-ip|node1 privte-ip
+# node2 public-ip|node2 privte-ip
+# ....
+# e.g.
+# 35.197.92.75|10.138.0.6
+# 35.185.232.222|10.138.0.3
+# 35.233.137.53|10.138.0.5
+
+nodes_file=$1
+remote_user=$2
+ssh_keys=$3
+
+declare -A ip_addr_array
+while IFS=\| read public private
+do
+  ip_addr_array[$public]=$private
+done < $nodes_file
+
+count=0
+leader=
+for public_ip in ${!ip_addr_array[*]}
+do
+  rsync -r -av ~/.cargo/bin $remote_user@$public_ip:~/.cargo
+  rsync -r -av ./multinode-demo $remote_user@$public_ip:~/solana/
+  
+  if [ -z $ssh_keys ]
```

Comment:
prefer [[ to [.  [[ is builtin and [ runs a command

Reply:
Can shellcheck be configured to enforce that?

Reply:
I don't think so